GAE/test/org/akvo/flow/api/app/DataPointServletTest.java

```diff
@@ -0,0 +1,214 @@
+            dataPoint.setDisplayName("dataPoint: " + i);
+            datapoints.add(dataPoint);
+        }
+        return new ArrayList<>(dpDao.save(datapoints));
+    }
+
+    private Set<Long> getEntityIds(List<? extends BaseDomain> entities) {
+        Set<Long> entityIds = new HashSet<>();
+        for (BaseDomain o : entities) {
+            entityIds.add(o.getKey().getId());
+        }
+        return entityIds;
+    }
+
+    @Test
+    public void datastoreReadWriteTest() {
+        final Long assignmentId = 1L;
+        final Long deviceId = 1L;
+        final List<Long> dataPointIds = Collections.singletonList(0L);
+        final Long surveyId = 1L;
+
+        createDataPointAssignment(assignmentId, deviceId, dataPointIds, surveyId);
+
+        final DataPointAssignmentDao dataPointAssignmentDao = new DataPointAssignmentDao();
+        final List<DataPointAssignment> dataPointAssignments = dataPointAssignmentDao.listByDeviceAndSurvey(deviceId, surveyId);
+        assertFalse(dataPointAssignments.isEmpty());
+
+        final DataPointAssignment assignment = dataPointAssignments.get(0);
+        assertEquals(dataPointIds, assignment.getDataPointIds());
+        assertEquals(deviceId, assignment.getDeviceId());
+        assertEquals(surveyId, assignment.getSurveyId());
+
+
+        final List<Long> deviceIds = Arrays.asList(1L, 2L);
+        final List<Long> formIds = Arrays.asList(100L, 200L);
+
+        createAssignment(surveyId, deviceIds, formIds);
+
+        final SurveyAssignmentDao saDao = new SurveyAssignmentDao();
+        final List<SurveyAssignment> surveyAssignments = saDao.listAllContainingDevice(2L);
+        assertFalse(surveyAssignments.isEmpty());
+
+        final SurveyAssignment sa = surveyAssignments.get(0);
+        assertEquals(surveyId, sa.getSurveyId());
+        assertTrue(sa.getDeviceIds().contains(2L));
+        assertEquals(formIds, sa.getFormIds());
+    }
+
+    @Test
+    public void someDataPointsTest() {
+
+        final Long surveyId = 100L;
+        final List<SurveyedLocale> dataPoints = createDataPoints(surveyId, 10);
+
+        final SurveyedLocale dataPoint = dataPoints.get(7);
+        assertEquals("7", dataPoint.getIdentifier());
+
+        final List<Long> selectedDataPointIds = Arrays.asList(dataPoints.get(5).getKey().getId(), dataPoints.get(6).getKey().getId());
+        final Long assignmentId = 20L;
+        final Long deviceId = 1L;
+
+        createDataPointAssignment(assignmentId, deviceId, selectedDataPointIds, surveyId);
+
+        final DataPointServlet servlet = new DataPointServlet();
+
+        final List<SurveyedLocale> someDataPoints = servlet.getDataPointList(surveyId, deviceId);
+
+        final Set<Long> selectedIds = new HashSet<>(selectedDataPointIds);
+        final Set<Long> foundIds = getEntityIds(someDataPoints);
+
+        assertEquals(selectedIds, foundIds);
+    }
+
+
+    /**
+     * New schema: `[0]` as data point list represent all data points
```

Comment:
This special value is not obvious where it is used in the tests. 

Also, try to remove comment by making the test say the same.

Reply:
I really dislike special values like [0]. They always cause problems. 

Did you discard the option of adding a flag in the survey assignment objects?

Reply:
We're storing `[0]` for _All data points_ as in the JDO/Dao writes/reads a List of ids (longs). Instead of storing a potential large List of Ids, we have a _constant_ that represent all _current_ and _future_ datapoints for a Survey.

Reply:
I feel that my question was not understood. Rephrasing it:

Did you discard the option of adding an explicit flag/field in SurveyAssignment object to mean "ALL DATAPOINTS" or "PARTIAL DATAPOINTS"?